tests/controllers/settings_change_second_password_ctrl_spec.coffee

```diff
@@ -79,32 +84,45 @@ describe "ManageSecondPasswordCtrl", ->
 
     it "should be valid if the password is ok", ->
       scope.form.password.$setViewValue('validpw')
+      $timeout.flush()
       expect(scope.form.password.$valid).toEqual(true)
 
     it "should reset the confirmations field when changed", ->
       scope.form.confirmation.$setViewValue('conf')
       scope.form.password.$setViewValue('validpw')
+      $timeout.flush()
       expect(scope.fields.confirmation).toEqual('')
 
     it "should not be valid if the password is too short", ->
       scope.form.password.$setViewValue('asd')
+      $timeout.flush()
       expect(scope.form.password.$valid).toEqual(false)
       expect(scope.form.password.$error.minlength).toEqual(true)
 
     it "should not be valid if the password is the password hint", ->
       scope.form.password.$setViewValue('passhint')
+      $timeout.flush()
       expect(scope.form.password.$valid).toEqual(false)
       expect(scope.form.password.$error.isValid).toEqual(true)
 
     it "should not be valid if the password is the main password", ->
       scope.form.password.$setViewValue('mainpw')
+      $timeout.flush()
+      expect(scope.form.password.$valid).toEqual(false)
+      expect(scope.form.password.$error.isValid).toEqual(true)
+
+    it "should not be valid if incorrect sec pass is entered when removing", ->
```

Comment:
This test fails on my local machine on the branch for #1090. Any idea what could have broken it?